src/commands/buffer.ts

```diff
@@ -37,19 +37,42 @@ export async function execute(
     let format: string[] = []
 
     for (const message of messages) {
+      limitMessageLength(message, 4000)
       format.push(`${formatMessage(message)}\n`)
-      if (format.join("").length > 4096) {
-        format.shift()
+      while (format.join("").length > 4000) {
+        removeLongestString(format)
       }
     }
-
     embed.setDescription(format.join(""))
   } else {
     embed.setDescription("No messages found.")
   }
   msg.reply({ embeds: [embed] })
 }
 
+function limitMessageLength(message: ChatBufferMessage, maxLength: number) {
+  if (message.content.length > maxLength) {
+    message.content = message.content.substring(0, maxLength)
+  }
+  return
+}
+
+function removeLongestString(stringArray: string[]) {
+  let maxLength = 0
+  let indexToRemove = -1
+
+  stringArray.forEach((str, index) => {
+    if (str.length > maxLength) {
+      maxLength = str.length
+      indexToRemove = index
+    }
+  })
+
+  if (indexToRemove !== -1) {
+    stringArray.splice(indexToRemove, 1)
```

Comment:
👀 this might mess with the ordering of messages, be careful

reasoning: it removes the largest string, so it can do something like:
```
NORMAL:

Alice: hello
Konpaku: [insert long rant here about life]
Alice: what..
```

```
PASSED:

Allice: hello
Allice: what..
```